server/src/main/java/org/elasticsearch/script/ScoreScript.java

```diff
@@ -104,10 +112,52 @@ public void setScorer(Scorable scorer) {
         };
     }
 
+    /**
+     * Accessed as _score in the painless script
+     * @return the score of the inner query
+     */
     public double get_score() {
         return scoreSupplier.getAsDouble();
     }
 
+    /**
+     * @return the internal document ID
+     */
+    public int getDocId() {
+        return docBase + docId;
+    }
+
+    /**
+     * @return shard id or throws an exception if shard is not set up for this script instance
+     */
+    public int getShardId() {
+        if (shardId > -1) {
+            return shardId;
+        } else {
+            throw new IllegalArgumentException("shard id can not be looked up!");
+        }
+    }
+
+    /**
+     * @return index name or throws an exception if the index name is not set up for this script instance
+     */
+    public String getIndex() {
+        if (indexName != null) {
+            return indexName;
+        } else {
+            throw new IllegalArgumentException("index name can not be looked up!");
+        }
+    }
+
+    public void setShard(int shardId) {
```

Comment:
I'm guessing we probably don't want the user setting the shard id in a script so maybe we can rename this to _setShard?  @rjernst What do you think here?